### DISCUSSION TITLE
How to deal with overfitting when using PyCaret with xgboost, catboost and lightgbm?

### DISCUSSION THREAD

**Discussion by garar** [pycaret/pycaret#3626] [Q&A]
Hello!

I'm training a simple classification mode using PyCaret. However I've noticed a lot of overfitting for some models (xgboost, catboost and lightgbm). Other models look good - similar performance for train, test and validation data sets. 

Is there a prescribed way to dealing with overfitting in PyCaret? I can do it in raw xgboost, but I like PyCaret and would like to continue using it.

Thanks!

**Comment by sharulhafiz** [comment]
How do you know your models overfit?

**Reply by garar** [reply]
Hi @sharulhafiz !

I've missed you comment. My model has nearly 100% accuracy on training data, and only 55% accuracy on validation data.

To quote https://towardsdatascience.com/handling-overfitting-in-deep-learning-models-c760ee047c6e :

```
We can identify overfitting by looking at validation metrics, like loss or accuracy. 
Usually, the validation metric stops improving after a certain number of epochs 
and begins to decrease afterward. The training metric continues to improve because 
the model seeks to find the best fit for the training data.
```

**Reply by sharulhafiz** [reply]
Thank you. I'm not new to ML but still have lots to learn. So how do you handle them in PyCaret?

**Reply by garar** [reply]
I don't know, this is why I've started this discussion :)
